### PR TITLE
Promote from BLAIS5-3123 to main

### PR DESCRIPTION
**Pull request by social-surveys-blaise-concourse**
Auto generated by concourse:

Promote from BLAIS5-3123 to main

### CODE REVIEW

File: src/components/QuestionnaireSelector.test.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                                  setInstruments={ setInstruments }
+                                  submitFunction={ submit }/>
+            </Router>
+        );
+    }
+
+    it("matches snapshot", async () => {
+        mockAdapter.onPost("/api/instruments").reply(200, instrumentDataReturned);
+        await act(async () => {
+            await flushPromises();
```
[the remark this answers is not in the document]
Comment (tomphp) [reply src/components/QuestionnaireSelector.test.tsx:64]: e.g replace with

```typescript
await screen.findByText('LMS2101_AA1');
```

File: src/components/QuestionnaireSelector.test.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+function expectAvailableQuestionnairesToBeDefined(wrapper: RenderResult, questionnairesReturned: string[]){
+    questionnairesReturned.forEach(function(questionnaire){
+        expect(wrapper.getByText(questionnaire)).toBeDefined();
+    })
+}
+
+function expectCheckboxesForEachQuestionnaireToBeDefined(wrapper: RenderResult, questionnairesReturned: string[]){
+    questionnairesReturned.forEach(function(questionnaire){
+        expect(wrapper.getByText(questionnaire)).toHaveClass("checkbox__label");
```
Comment (tomphp) [inline src/components/QuestionnaireSelector.test.tsx:124]: Consider `getByRole('checkbox', { ... })`

File: src/components/QuestionnaireSelector.test.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    })
+}
+function expectTickWhenAQuestionnaireIsSelected(wrapper: RenderResult){
+    const checkBox = wrapper.getByText("LMS2101_AA1");
+    fireEvent.click(checkBox);
+    expect(checkBox).toBeChecked;
+}
+
+async function expectSelectAllToSelectAllQuestionnaires(wrapper: RenderResult, questionnairesReturned: string[]){
+    await act(async () => {
```
Comment (tomphp) [inline src/components/QuestionnaireSelector.test.tsx:134]: no need for `act`

File: src/components/QuestionnaireSelector.test.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+function expectCheckboxesForEachQuestionnaireToBeDefined(wrapper: RenderResult, questionnairesReturned: string[]){
+    questionnairesReturned.forEach(function(questionnaire){
+        expect(wrapper.getByText(questionnaire)).toHaveClass("checkbox__label");
+    })
+}
+function expectTickWhenAQuestionnaireIsSelected(wrapper: RenderResult){
+    const checkBox = wrapper.getByText("LMS2101_AA1");
+    fireEvent.click(checkBox);
+    expect(checkBox).toBeChecked;
```
Comment (tomphp) [inline src/components/QuestionnaireSelector.test.tsx:130]: Should this have `()`?

File: src/components/QuestionnaireSelector.test.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    await act(async () => {
+        fireEvent.click(await screen.findByText(/Select all/));
+    });
+    questionnairesReturned.forEach((questionnaire) => {
+        expect(wrapper.getByText(questionnaire)).toBeChecked();
+    })
+}
+
+async function expectSelectAllToChangeToUnselectAll(wrapper: RenderResult) {
+    await act(async () => {
```
Comment (tomphp) [inline src/components/QuestionnaireSelector.test.tsx:143]: `act` again

File: src/components/QuestionnaireSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            return getInstrumentList(surveyTla, interviewer, startDate, endDate);
+        }
+
+        fetchInstruments()
+            .then(setupForm)
+            .catch((error: Error) => {
+                setStatus("loading_failed");
+                console.error(`Response: Error ${error}`);
+            });
+            return () => {
```
Comment (tomphp) [inline src/components/QuestionnaireSelector.tsx:57]: indented too far

**Review by tomphp — CHANGES_REQUESTED** [review]

**Review by JAWilliamsONS — APPROVED** [review]
looks good to me :) visual inspection of the code looks good
Deployed to my sandbox, tests passed and functions as expected

**Review by JAWilliamsONS — APPROVED** [review]

**Review by elthorne — APPROVED** [review]

**Pull request merged**